javascript/javascript1/week4/voice-assistant.js

```diff
@@ -0,0 +1,275 @@
+    return `You have ${user.todo.length} todos:\n${todoItems}`;
+  }
+}
+
+function getCurrentDate() {
+  const today = new Date();
+  const options = {
+    day: 'numeric',
+    month: 'long',
+    year: 'numeric',
+  };
+  const formattedDate = today.toLocaleDateString('en-GB', options); //15 May 2024
+  const [day, month, year] = formattedDate.split(' ');
+  return `Today is ${day}. of ${month} ${year}`;
+}
+
+function calcMathExpression(mathExpression) {
+  if (isValidMathExpression(mathExpression)) {
+    return eval(mathExpression).toString();
+  } else {
+    return 'I am afraid I did not catch the numbers. Could you repeat please ';
+  }
+}
+
+function setTimer(timeInfo, command) {
+  //calculating milliseconds for a timer
+  const [timeValue, timeUnit] = timeInfo;
+  let timeInMilliseconds;
+  switch (timeUnit) {
+    case 'seconds':
+    case 'second':
+      timeInMilliseconds = timeValue * 1000;
+      break;
+    case 'minutes':
+    case 'minute':
+      timeInMilliseconds = timeValue * 60 * 1000;
+      break;
+    case 'hours':
+    case 'hour':
+      timeInMilliseconds = timeValue * 60 * 60 * 1000;
+      break;
+    default:
+      return 'Oh, I did not catch the time. Please use seconds, minutes, or hours.'; // Return for invalid time units
+  }
+
+  // Clear existing timer
+  if (user.activeTimerId !== null) {
+    clearTimeout(user.activeTimerId);
+  }
+  // Set the new timer
+  user.activeTimerId = setTimeout(() => {
+    console.log(getReply(command)); //for seeing the message in consol
+    //code for working artyom.js
+    // const response = getReply(command);
+    // artyom.say(response);
+    clearTimeout(user.activeTimerId);
+  }, timeInMilliseconds);
+
+  return `Timer set for ${timeValue} ${timeUnit}`;
+}
+
+function getNiceMessage() {
+  const nicePhrases = [
+    'You are an incredible person',
+    'You are a wonderful friend',
+    'Your positivity is infectious',
+    'You have a great sense of humor',
+    'You are a great coder',
+    'Regular expressions are a piece of cake for you',
+    'You will soon be working as a web developer',
+  ];
+  const randomIndex = Math.floor(Math.random() * nicePhrases.length);
+  return `${nicePhrases[randomIndex]}, ${user.name}`;
+}
+
+//functions for recognizing a command
+const isSayHello = (command) => command.includes('hello my name is');
+const askName = (command) => command.includes('what is my name');
+const addTodo = (command) =>
+  command.startsWith('add') && command.endsWith('to my to-do');
+const removeTodo = (command) =>
+  command.startsWith('remove') && command.endsWith('from my to-do');
+const askListTodos = (command) => command.includes('what is on my to-do');
+const askCurrentDate = (command) => command.includes('what day is it today');
+const askDoingMath = (command) =>
+  command.startsWith('what is') &&
+  (command.includes('+') ||
+    command.includes('plus') ||
+    command.includes('-') ||
+    command.includes('minus') ||
```

Comment:
I saw others had to do the same - use both symbol and word, otherwise you never know how the script will hear you.

Reply:
Exactly!